### PR TITLE
Enh/share expiration notification

### PR DESCRIPTION
**Pull request by rullzer**
Depends on:
- [x] https://github.com/nextcloud/server/pull/17739

Todo:
- [x] Rebase on master once https://github.com/nextcloud/server/pull/17739 is in
- [X] Show actual notification
  - [x] Cleanup notifiations

### CODE REVIEW

File: apps/files_sharing/lib/BackgroundJob/ExiprationNotificationJob.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+			);
+
+		$shares = $qb->execute();
+		$now = $this->time->getDateTime();
+		while($share = $shares->fetch()) {
+			$notification = $this->notificationManager->createNotification();
+			$notification->setApp('files_sharing')
+				->setDateTime($now)
+				->setObject('share', (string)$share['id'])
+				->setSubject('expiresTomorrow');
```
Comment (nickvergessen) [inline apps/files_sharing/lib/BackgroundJob/ExiprationNotificationJob.php]: this is not handled in this PR?

Comment (rullzer) [reply apps/files_sharing/lib/BackgroundJob/ExiprationNotificationJob.php]: No not yet

**Comment by rullzer** [comment]
As discussed lets move this to a command that can be run

**Comment by rullzer** [comment]
Moved it to a command you can now run once a day or so from cron.

**Comment by tcitworld** [comment]
> As discussed lets move this to a command that can be run

Why not make like #3044, one mode with background jobs and one with command?

**Comment by rullzer** [comment]
@tcitworld because right now I don't want to have this by default. Only systems that want to have this will have to run the command.

Also since we can't gurantee whena background job is run this creates some issues right now ;)

**Comment by rullzer** [comment]
Rebased, review tine

File: apps/files_sharing/lib/Command/ExiprationNotification.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	protected function configure() {
+		$this
+			->setName('sharing:expiration-notification')
+			->setDescription('Notify share initiators when a share will expire the next day.');
+	}
+
+	public function execute(InputInterface $input, OutputInterface $output) {
+		//Current time
+		$minTime = $this->time->getDateTime();
+		$minTime->add(new \DateInterval('P1D'));
```
Comment (skjnldsv) [inline apps/files_sharing/lib/Command/ExiprationNotification.php]: So we always use the server time to expire, right?
We don't store any utc shift?

**Review by skjnldsv — APPROVED** [review]
Code looks good :)

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**